### PR TITLE
knife's edge test classic pomdps

### PR DESCRIPTION
**Pull request by atkirtland**

### CODE REVIEW

File: scripts/knife_edge/test_ld_zero_conditions.py [hunk 1]
```diff
@@ -193,22 +207,64 @@ def get_K(Pi_s, td_policy, lambda_):
...
-        pi = get_policy(spec, *probs)
+        if n_params > 0:
+            pi = get_policy(spec, *probs)
+        else:
+            pi = get_policy(spec)
+            assert isinstance(pi, str) and pi == "random"
+            n_o = pomdp.phi.shape[1]
+            n_a = pomdp.T.shape[0]
+            pi = np.random.rand(n_o, n_a)
+            pi = pi / pi.sum(axis=1)[:, None]
```
Comment (camall3n) [inline scripts/knife_edge/test_ld_zero_conditions.py]: Can we move this into the `get_policy()` function? Oh, I guess it needs a size. Can we make a `get_random_policy(size)` function?

File: scripts/knife_edge/test_ld_zero_conditions.py [hunk 2]
```diff
@@ -74,6 +74,20 @@ def get_policy(spec, p, q=0.5, r=0.5, a=0.5):
...
+        "example_7",
+        "4x3.95",
+        "cheese.95",
+        "network",
+        "shuttle.95",
+        "paint.95",
+        "hallway",
+        "bridge-repair",
+    ]:
+        pi = "random"
```
Comment (camall3n) [inline scripts/knife_edge/test_ld_zero_conditions.py]: This isn't a policy, so it's the wrong return type, which will lead to us being confused in the future when we forget what's happening.

File: scripts/knife_edge/test_ld_zero_conditions.py [hunk 3]
```diff
@@ -193,22 +207,64 @@ def get_K(Pi_s, td_policy, lambda_):
...
+            "∆ Q_sa",
+            "∆ Q_wa",
+            "Δ SF_oasa",
+            "Δ SF_oaoa_yellow_red",
+            "Δ SF_oaoa_blue_red",
+            "Δ SF_TD_yellow_blue",
+        ]
+    ]
+    .sort_index(key=spec_order)
+)
```
Comment (camall3n) [inline scripts/knife_edge/test_ld_zero_conditions.py:254]: This is way prettier 😍

**Review by camall3n — COMMENTED** [review]
I'm wondering if it makes sense to turn this into a separate script, or to refactor into separate modes.

File: scripts/knife_edge/test_ld_zero_conditions.py (hunk 1 above)
Comment (atkirtland) [reply scripts/knife_edge/test_ld_zero_conditions.py]: Done!

File: scripts/knife_edge/test_ld_zero_conditions.py (hunk 2 above)
Comment (atkirtland) [reply scripts/knife_edge/test_ld_zero_conditions.py]: The call to this function was actually unnecessary for the additional classic POMDPs, so I just removed it.

**Comment by atkirtland** [comment]
I think it's cleaner to just have it in one file as it is a strict generalization of the knife's edge POMDP case. Also, I don't think this file will be used enough to justify modularizing the random policy code and separating it into another file.

**Comment by camall3n** [comment]
Thanks for making these changes! Looks good!

**Pull request merged**